eth/downloader/downloader_test.go

```diff
@@ -1541,3 +1530,42 @@ func testFakedSyncProgress(t *testing.T, protocol int, mode SyncMode) {
 		t.Fatalf("Final progress mismatch: have %v/%v/%v, want 0-%v/%v/%v", origin, current, latest, targetBlocks, targetBlocks, targetBlocks)
 	}
 }
+
+// This test reproduces an issue where unexpected deliveries would
+// block indefinitely if they arrived at the right time.
+func TestDeliverHeadersHang62(t *testing.T) {
```

Comment:
Please change this test so that it's run for 62, 63 and 64 too, similar to other rest of the tests in the suite. The various protocols use slightly different paths in the code so it's important to have them all tested and passing.